src/main/java/gov/cabinetoffice/gap/adminbackend/security/interceptors/AuthorizationHeaderInterceptor.java

```diff
@@ -1,50 +1,91 @@
     public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) {
         log.info("Intercepting request: " + request.getRequestURI());
-        if (handler instanceof final HandlerMethod handlerMethod) {
+        if (handler instanceof HandlerMethod handlerMethod) {
             final Method method = handlerMethod.getMethod();
 
-            SpotlightPublisherHeaderValidator annotation = method
-                    .getAnnotation(SpotlightPublisherHeaderValidator.class);
-            log.info("SpotlightPublisherHeaderValidator: " + annotation);
+            final LambdasHeaderValidator annotation = method.getAnnotation(LambdasHeaderValidator.class);
 
             if (annotation != null) {
+                log.info("Request is coming from lambda, validating authorization header");
+
                 final String authorizationHeader = request.getHeader(HttpHeaders.AUTHORIZATION);
-                if (authorizationHeader == null || !authorizationHeader.equals(expectedAuthorizationValue)) {
-                    // TODO do we want this log?? is it safe?? shall we change what it
-                    // says??
+                final boolean isAuthorizationHeaderCorrect = compareAuthorizationSecret(authorizationHeader,
+                        expectedAuthorizationValue, privateKey);
+                if (authorizationHeader == null || !isAuthorizationHeaderCorrect) {
+
                     log.info("Authorization Header Value: " + authorizationHeader
                             + " does not match the expected value");
 
                     response.setStatus(HttpServletResponse.SC_FORBIDDEN);
                     return false;
                 }
+
+                log.info("Authorization Header Value matches the expected value");
             }
         }
 
         return true;
     }
 
+    private boolean compareAuthorizationSecret(String authorizationHeader, String expectedAuthorizationValue,
+            String privateKey) {
+        return decrypt(authorizationHeader, privateKey).equals(expectedAuthorizationValue);
+
+    }
+
+    private String decrypt(String encryptedText, String privateKeyString) {
+        try {
+
+            final byte[] privateKeyBytes = Base64.getDecoder().decode(privateKeyString);
+            final byte[] encryptedMessageBytes = Base64.getDecoder().decode(encryptedText);
+
+            final PKCS8EncodedKeySpec keySpec = new PKCS8EncodedKeySpec(privateKeyBytes);
+            final KeyFactory keyFactory = KeyFactory.getInstance("RSA");
+            final PrivateKey rsaPrivateKey = keyFactory.generatePrivate(keySpec);
+
+            final Cipher decryptCipher = Cipher.getInstance("RSA");
+            decryptCipher.init(Cipher.DECRYPT_MODE, rsaPrivateKey);
+
+            final byte[] decryptedMessageBytes = decryptCipher.doFinal(encryptedMessageBytes);
+
+            return new String(decryptedMessageBytes, StandardCharsets.UTF_8);
+        }
+        catch (Exception e) {
```

Comment:
Could we specify one or more exception types here and be more specific rather than catching everything? Do any of the methods called here have specific throws declarations?

Reply:
good shout, done